### PR TITLE
BUILD(vcpkg) Fix --clean-after-build issue affecting qt5-base

### LINKED ISSUE #4674: vcpkg script for Windows fails with -CleanAfterBuild $true
**Describe the bug**
When running the vcpkg script for Windows with `-CleanAfterBuild $true` the script will fail while building `qt5-base`

**Steps to Reproduce**
1. Run: `./mumble/scripts/vcpkg/get_mumble_dependencies.ps1 -CleanAfterBuild $true`
2. The script has a build error and exits attempting to build `qt5-base`

**Expected behavior**
The script should run successfully

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Version: 1.4.0

**Additional context**
The issue is related to the dependency `icu` which handles Unicode for Windows. An issue has been opened with the vcpkg team:
https://github.com/microsoft/vcpkg/issues/15488

### PR DESCRIPTION
**Pull request by ZeroAbility**
Modify function to accept a string array param
Add mdnsresponder to the packages array
Remove individual package installs affected by previous --clean-after-build issue
Remove iterator that installs packages independently
Remove unneeded double quotes for variables
Fixes #4674

### CODE REVIEW

File: scripts/vcpkg/get_mumble_dependencies.ps1 [hunk 1]
```diff
@@ -6,29 +6,29 @@ $mumble_deps = "qt5-base",
                "qt5-tools",
                "grpc",
                "boost-accumulators",
-               "opus",
```
Comment (Krzmbrzl) [inline scripts/vcpkg/get_mumble_dependencies.ps1:9]: Why did you remove `opus` and how is that related to fixing the issue at hand? :thinking:

File: scripts/vcpkg/get_mumble_dependencies.ps1 [hunk 2]
```diff
@@ -6,29 +6,29 @@ $mumble_deps = "qt5-base",
...
+               "mdnsresponder",
+               "zlib", 
                "zeroc-ice"
 
 $ErrorActionPreference = 'Stop'
 
 function vcpkg_install {
 	Param(
-		[string] $package,
+		[string[]] $package,
```
Comment (Krzmbrzl) [inline scripts/vcpkg/get_mumble_dependencies.ps1:23]:
```suggestion
		[string[]] $packages,

```

File: scripts/vcpkg/get_mumble_dependencies.ps1 [hunk 3]
```diff
@@ -6,29 +6,29 @@ $mumble_deps = "qt5-base",
...
 		[string] $targetTriplet,
 		[switch] $cleanAfterBuild = $false
 	)
 	
 	if ($cleanAfterBuild) {
-		./vcpkg.exe install "$package" --triplet "$targetTriplet" --clean-after-build
+		./vcpkg.exe install $package --triplet $targetTriplet --clean-after-build
```
Comment (Krzmbrzl) [inline scripts/vcpkg/get_mumble_dependencies.ps1:29]:
```suggestion
		./vcpkg.exe install $packages --triplet $targetTriplet --clean-after-build

```

File: scripts/vcpkg/get_mumble_dependencies.ps1 [hunk 4]
```diff
@@ -6,29 +6,29 @@ $mumble_deps = "qt5-base",
...
 	} else {
-		./vcpkg.exe install "$package" --triplet "$targetTriplet"
+		./vcpkg.exe install $package --triplet $targetTriplet
```
Comment (Krzmbrzl) [inline scripts/vcpkg/get_mumble_dependencies.ps1:31]:
```suggestion
		./vcpkg.exe install $packages --triplet $targetTriplet

```

File: scripts/vcpkg/get_mumble_dependencies.ps1 [hunk 5]
```diff
@@ -6,29 +6,29 @@ $mumble_deps = "qt5-base",
...
                "boost-accumulators",
-               "opus",
                "poco",
                "libvorbis",
                "libogg",
                "libflac",
                "libsndfile",
                "libmariadb",
-               "zlib",
+               "mdnsresponder",
```
Comment (Krzmbrzl) [inline scripts/vcpkg/get_mumble_dependencies.ps1:15]: Why was that not needed before? Isn't that a dependency only? :eyes:

**Review by Krzmbrzl — COMMENTED** [review]
The commit message does not quite follow our guidelines. The line `Fixes #4674` should be the last one (separated by a newline before).

Furthermore you duplicated the word "Fix" in the commit title. That's of course only a small thing but in order to fulfill the spirit of keeping the title as short as possible, it should be removed.

And finally (though that is probably subjective) I usually use e.g. `BUILD(cmake): Fix ...` instead of `FIX(build): ...`. Thus the TYPE should indicate that this is about a change to the build-process whereas FIX is usually used for bug fixes in code :shrug: 

Thus all in all my suggestion would be
```
BUILD(vcpkg) Fix --clean-after-build issue affecting qt5-base

The following changes were performed:
- Modify function to accept a string array param
- Add mdnsresponder to the packages array
- Remove individual package installs affected by previous --clean-after-build issue
- Remove iterator that installs packages independently
- Remove unneeded double quotes for variables

Fixes #4674
```

File: scripts/vcpkg/get_mumble_dependencies.ps1 (hunk 1 above)
Comment (ZeroAbility) [reply scripts/vcpkg/get_mumble_dependencies.ps1:9]: We are using a submodule to deliver opus. I can add it back if needed, but it's just occupying space.

File: scripts/vcpkg/get_mumble_dependencies.ps1 (hunk 5 above)
Comment (ZeroAbility) [reply scripts/vcpkg/get_mumble_dependencies.ps1:15]: I don't recall removing zlib, so it must have been accidental.

**Comment by ZeroAbility** [comment]
> The commit message does not quite follow our guidelines. The line `Fixes #4674` should be the last one (separated by a newline before).
> 
> Furthermore you duplicated the word "Fix" in the commit title. That's of course only a small thing but in order to fulfill the spirit of keeping the title as short as possible, it should be removed.
> 
> And finally (though that is probably subjective) I usually use e.g. `BUILD(cmake): Fix ...` instead of `FIX(build): ...`. Thus the TYPE should indicate that this is about a change to the build-process whereas FIX is usually used for bug fixes in code 
> 
> Thus all in all my suggestion would be
> 
> ```
> BUILD(vcpkg): Fix --clean-after-build issue affecting qt5-base
> 
> The following changes were performed:
> - Modify function to accept a string array param
> - Add mdnsresponder to the packages array
> - Remove individual package installs affected by previous --clean-after-build issue
> - Remove iterator that installs packages independently
> - Remove unneeded double quotes for variables
> 
> Fixes #4674
> ```

I will get this updated.

File: scripts/vcpkg/get_mumble_dependencies.ps1 (hunk 5 above)
Comment (ZeroAbility) [reply scripts/vcpkg/get_mumble_dependencies.ps1:15]: mdnsresponder was installed without `--clean-after-build` before when we were doing each package individually. It was moved to the array.

File: scripts/vcpkg/get_mumble_dependencies.ps1 (hunk 1 above)
Comment (Krzmbrzl) [reply scripts/vcpkg/get_mumble_dependencies.ps1:9]: True ^^

This change is however unrelated to the actual fix, so I'd prefer having that in a separate commit :)

File: scripts/vcpkg/get_mumble_dependencies.ps1 (hunk 5 above)
Comment (Krzmbrzl) [reply scripts/vcpkg/get_mumble_dependencies.ps1:15]: I just saw that we used to install `icu` as well but this doesn't seem to have been moved up here. Is this just not needed?

Comment (ZeroAbility) [reply scripts/vcpkg/get_mumble_dependencies.ps1:15]: `qt5-base` marks it during install.

File: scripts/vcpkg/get_mumble_dependencies.ps1 (hunk 1 above)
Comment (ZeroAbility) [reply scripts/vcpkg/get_mumble_dependencies.ps1:9]: This has been done.

**Review by Krzmbrzl — APPROVED** [review]
